src/scripts/main.js

```diff
@@ -1,5 +1,32 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
+const wall = document.querySelector('.wall');
+
+const spider = document.querySelector('.spider');
+
+const wallParam = wall.getBoundingClientRect();
+
+wall.addEventListener('click', (e) => {
   // write code here
+  let left = Math.min(
+    e.clientX - wall.offsetLeft - spider.clientWidth / 2 - wall.clientLeft,
+    wallParam.right -
+      wall.offsetLeft -
+      2 * wall.clientLeft -
+      spider.clientWidth,
+  );
+
+  let topS = Math.min(
+    e.clientY - wall.offsetTop - spider.clientHeight / 2 - wall.clientTop,
+    wallParam.bottom -
+      wall.offsetTop -
+      2 * wall.clientTop -
+      spider.clientHeight,
+  );
```

Comment:
Similarly, the calculation for 'topS' ensures that the spider does not move outside the bottom boundary of the wall. Consider checking if 'e.clientY' is less than 'wall.offsetTop' to prevent the spider from moving outside the top boundary.